### PR TITLE
Poll Provider for acceptance only till (deal start epoch + grace period) has elapsed 

### PR DESCRIPTION
**Pull request by aarshkshah1992**
Closes https://github.com/filecoin-project/go-fil-markets/issues/503.

### CODE REVIEW

File: storagemarket/impl/clientstates/client_states.go [hunk 1]
```diff
@@ -23,6 +23,10 @@ import (
 
 var log = logging.Logger("storagemarket_impl")
 
+// MaxGraceEpochsForDealAcceptance is the maximum number of epochs we will wait for post the start epoch
```
Comment (dirkmc) [inline storagemarket/impl/clientstates/client_states.go:26]:
```suggestion
// MaxGraceEpochsForDealAcceptance is the maximum number of epochs we will wait for past the start epoch
```

File: storagemarket/impl/clientstates/client_states.go [hunk 2]
```diff
@@ -170,6 +174,13 @@ func InitiateDataTransfer(ctx fsm.Context, environment ClientDealEnvironment, de
 
 // CheckForDealAcceptance is run until the deal is sealed and published by the provider, or errors
 func CheckForDealAcceptance(ctx fsm.Context, environment ClientDealEnvironment, deal storagemarket.ClientDeal) error {
+	_, currEpoch, err := environment.Node().GetChainHead(ctx.Context())
+
+	if err == nil {
+		if currEpoch > deal.Proposal.StartEpoch+MaxGraceEpochsForDealAcceptance {
+			return ctx.Trigger(storagemarket.ClientEventDealRejected, deal.State, "start epoch already elapsed ")
```
Comment (dirkmc) [inline storagemarket/impl/clientstates/client_states.go:181]:
```suggestion
			return ctx.Trigger(storagemarket.ClientEventDealRejected, deal.State, "start epoch already elapsed")
```

**Review by dirkmc — APPROVED** [review]
Looks great 👍 

(two small nits)

**Comment by aarshkshah1992** [comment]
@dirkmc Have accepted your suggestions. Please merge this when you want to.

**Comment by s0nik42** [comment]
How can I know if this is implemented in a specific version of lotus ?

**Comment by dirkmc** [comment]
@s0nik42 this was included in [release v1.2.4](https://github.com/filecoin-project/go-fil-markets/pull/524) which is in v1.9.0-rc3 of lotus:
https://github.com/filecoin-project/lotus/blob/v1.9.0-rc3/go.mod#L37

**Comment by s0nik42** [comment]
Thank you for pointing me to this !

**Pull request merged**